scripts/sanitize-mocha-results.js

```diff
@@ -18,7 +18,7 @@ function isAllowlistedEnv (key, value) {
     ['true', 'false', 'TRUE', 'FALSE', 'True', 'False'].includes(value)
     || value.length < 4
     // allow some envs that are not sensitive
-    || ['nodejs_version', 'CF_DOMAIN', 'SKIP_RELEASE_CHANGELOG_VALIDATION_FOR_BRANCHES', 'CIRCLE_PROJECT_REPONAME', 'CIRCLE_PR_REPONAME', 'CIRCLE_BRANCH',
+    || ['nodejs_version', 'CF_DOMAIN', 'SKIP_RELEASE_CHANGELOG_VALIDATION_FOR_BRANCHES', 'CIRCLE_PROJECT_REPONAME', 'CIRCLE_PR_REPONAME', 'CIRCLE_BRANCH', 'CIRCLE_PROJECT_USERNAME',
```

Comment:
```suggestion
    || ['nodejs_version', 'CF_DOMAIN', 'SKIP_RELEASE_CHANGELOG_VALIDATION_FOR_BRANCHES', 'CIRCLE_PROJECT_REPONAME', 'CIRCLE_PR_REPONAME', 'CIRCLE_BRANCH', 
```

Unless we need to keep this env in the mocha output!

Reply:
@cacieprins I added because I saw this in error the circle output while finalizing the `unit-tests` - https://app.circleci.com/pipelines/github/cypress-io/cypress/71555/workflows/b5f2dd2f-9bab-4e61-bb11-1606f93dedfb/jobs/2938779

```sh
Checking that test-results.bfde9f904e59c4d970df8e90a2d0d2e7.xml doesn't contains any environment variables...
Report parsed successfully.
Checking that test-results.c3ce4089ffd3753341b53984fcc3d5f0.xml doesn't contains any environment variables...
node:internal/process/promises:391
    triggerUncaughtException(err, true /* fromPromise */);
    ^

Error: Report contained the value of `CIRCLE_PROJECT_USERNAME`, which is a CI environment variable. This means that a failing test is exposing environment variables. Test reports will not be persisted for this job.
    at checkReportFile (/root/cypress/scripts/sanitize-mocha-results.js:53:11)
```

Since `CIRCLE_PROJECT_USERNAME` should just be (AFAIK) just `cypress-io`, it didn't seem like there was a reason we needed to have it in the list of env vars that are considered secrets

Reply:
Looks like that may have been related to test failures on the previous step, but adding this env to the allow list may help expose more details about that failure if it occurs again. 👍